core/src/main/java/com/exadel/etoolbox/linkinspector/api/service/LinkTypeProvider.java

```diff
@@ -0,0 +1,24 @@
+package com.exadel.etoolbox.linkinspector.api.service;
+
+import com.exadel.etoolbox.linkinspector.api.dto.LinkStatus;
+
+public interface LinkTypeProvider {
```

Comment:
(naming) The class which is named "LinkTypeProvider" is expected to provide something. What does it provide? By the name, it provides "link type". But it's not to the point. It provides two other things: the string value of the link, and then the link status. I think it is "LinkBroker", or "LinkManager", or simply "LinkProvider" (but then again, it does not only provide links, it provides _statuses_)